metis-dereference/metis-dereference-service/src/main/java/eu/europeana/metis/dereference/service/MongoDereferenceService.java

```diff
@@ -98,25 +110,33 @@ private EnrichmentBase retrieveCachedEntity(String resourceId)
     return result;
   }
 
-  private String retrieveFromSourceVocabulary(String resourceId) throws TransformerException {
+  private String retrieveFromSourceVocabulary(String resourceId)
```

Comment:
Can this method be split to have less return statements?
Perhaps one method could be to get the vocabulary and another to actually to the transformation using that vocabulary?

Reply:
Fixed.